### PR TITLE
auto detect line endings

### LINKED ISSUE #87: Auto detect newline characters
Heya! Flynt is amazing, thanks for making it!

It appears flynt writes files with the OS specific newline characters

This is a minor inconvenience when using flynt on a code base where files are saved with unix line endings on a computer running Windows and there's a need to retain the original line endings

Maybe flynt could auto detect newline characters of a file

### PR DESCRIPTION
**Pull request by Inveracity**
This PR adds a new function for detecting the newline character so that the files written back after changes retain the line endings the original file had.

closes #87

### CODE REVIEW

File: src/flynt/api.py [hunk 1]
```diff
@@ -82,7 +82,8 @@ def default_result():
         for l in unified_diff(contents.split('\n'), new_code.split('\n'), fromfile=filename):
             print(l)
     else:
-        with open(filename, "w", encoding="utf-8") as f:
+        ln = _auto_detect_line_ending(filename)
```
Comment (ikamensh) [inline src/flynt/api.py:85]: Could we use already-read `contents` of the file instead of reading it again?

**Comment by ikamensh** [comment]
Hey, thanks for your PR! I don't see any serious problems with it and will be happy to merge it. There's one question I got - I see that you read entire file one more time (for every file). Does it have a noticeable performance impact? This should be trivial to avoid because the function you have changed already has read the original file into memory. See my inline comment.

File: src/flynt/api.py (hunk 1 above)
Comment (Inveracity) [reply src/flynt/api.py]: Good point! 😄

**Comment by Inveracity** [comment]
Total brainfart on my part, of course it shouldn't read it twice 😄 
It should be all fixed now!

**Comment by ikamensh** [comment]
I am being diligently paranoid and want to ask how did you integration-test it? I see that detecting the correct line ending is working, but this doesn't fully imply that it will do what is expected end-to-end. Do you think you could add a test, e.g. similar to the ones in `test/integration/test_api.py` to simulate actually reformatting a file with different endings? The pipeline should automatically run test on both a linux box and a windows box.

Also maybe this is not normal, but could a single file have mixed line endings for a legit reason :D ?

**Comment by Inveracity** [comment]
I just discovered that doing `newline=""` forces the write to "universal" mode which writes the file with unaltered newlines. This is the actual fix. 

Then you don't have to worry about whether people have mixed newlines 😂

**Comment by ikamensh** [comment]
Haha, it's beautiful that in the end it's 1 line change. Thanks!!

**Comment by ikamensh** [comment]
Uploaded as version 0.59

**Pull request merged**